### PR TITLE
update cilium to version 0.15.0

### PR DESCRIPTION
**Pull request by msherif1234**
**- What this PR does and why is it needed**
update cilium to v0.15.0

this PR update cilium pkg as dependabot atm is stuck because of the dependency on go1.22 once that dependency is addressed dependabot will be doing this update in the future

go1.22 is blocked by https://github.com/openshift-metal3/dev-scripts/pull/1645

### CODE REVIEW

File: hack/generators.Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM fedora:37
+FROM fedora:40
```
Comment (frobware) [inline hack/generators.Dockerfile:1]: Is the bump here required because of the Cilium bump?

Comment (msherif1234) [reply hack/generators.Dockerfile:1]: no this is independent but while I was updating the cilium pkg I updated the os to pickup newer version of clang compiler

Comment (frobware) [reply hack/generators.Dockerfile:1]: Does cilium 0.15 require a newer version of clang? I was just trying to understand whether the Fedora 37->40 bump was a necessity.

Comment (msherif1234) [reply hack/generators.Dockerfile:1]: no it doesn't both updates are independent I just combined to avoid the another PR overhead

Comment (frobware) [reply hack/generators.Dockerfile:1]: /lgtm

**Pull request merged**